ecommerce/models.py

```diff
@@ -495,6 +510,12 @@ def fulfill(self, payment_data):
             mode=EDX_ENROLLMENT_VERIFIED_MODE,
         )
 
+        # record all the courserun in the order
+        for run in self.purchased_runs:
```

Comment:
A few things we could improve overall:

 - Move this code above the `create_run_enrollments` call so that if that API call to openedx fails for some reason we don't get in an inconsistent state.
 - Could you create an abstract base class that subclasses `Order` and that this proxy class would the subclass?
   - Move the `self.transaction.create` and `PaidCourseRun.objects.get_or_create` into their own dedicated functions there
   - Move the `create_run_enrollments()` call into its own dedicated function
   - Move the implementation of `fulfill()` there as well so we don't have duplicate code